src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/OpDelegationStrategy.cs

```diff
@@ -77,27 +77,28 @@ private bool IsSupportedNode(TexlNode node, OperationCapabilityMetadata metadata
             Contracts.AssertValue(binding);
             Contracts.AssertValue(opDelStrategy);
 
-            if (!binding.IsRowScope(node))
+            // Check whether this is -
+            //  1) in operator delegation and
+            //  2) it is verifying if RHS node is supported and
+            //  3) it is not an async node and
+            //  4) it is a single column table and
+            //  5) metadata belongs to cds datasource that supports delegation of CdsIn
+            // If this check fails, verify if it is simply a valid node..
+            // Eg of valid delegation functions -
+            // Filter(Accounts, 'Account Name' in ["Foo", Bar"]) - Direct table use
+            // Set(Names, ["Foo", Bar"]); Filter(Accounts, 'Account Name' in Names) - Using variable of type table
+            // ClearCollect(Names, Accounts); Filter(Accounts, 'Account Name' in Names.'Account Name') - using column from collection.
+            // This won't be delegated - Filter(Accounts, 'Account Name' in Accounts.'Account Name') as Accounts.'Account Name' is async.
+            if (binding.Document.Properties.EnabledFeatures.IsEnhancedDelegationEnabled &&
+                isRHSNode && (opDelStrategy as BinaryOpDelegationStrategy)?.Op == BinaryOp.In && !binding.IsAsync(node) && binding.GetType(node).IsTable && binding.GetType(node).IsColumn &&
```

Comment:
>(opDelStrategy as BinaryOpDelegationStrategy)?.Op == BinaryOp.I [](http://example.com/codeflow?start=29&length=63)

Shouldn't this be part of InOpDelegationStrategy if it is "in" operator specific?